### DISCUSSION TITLE
User authentication for emqx

### DISCUSSION THREAD

**Discussion by raheem071989** [emqx/emqx#13002] [Q&A] [answered]
Dear Team,

Could you please help in enabling user authentication here for wss and ssl configuration on port 8883 and 8084.

how to add username and password for this

cat emqx.conf

#listeners.tcp.default {

bind = "0.0.0.0:1883"
max_connections = 1024000
#}

listeners.ssl.default {
bind = "0.0.0.0:8883"
max_connections = 1024000
ssl_options {
cacertfile = "/etc/emqx/certs/int.pem"
certfile = "/etc/emqx/certs/ssl.pem"
keyfile = "/etc/emqx/certs/sslkey.pem"
verify = verify_none
fail_if_no_peer_cert = false
}
}

listeners.wss.default {
bind = "0.0.0.0:8084"
max_connections = 1024000
websocket.mqtt_path = "/mqtt"
ssl_options {
cacertfile = "/etc/emqx/certs/int.pem"
certfile = "/etc/emqx/certs/ssl.pem"
keyfile = "/etc/emqx/certs/sslkey.pem"
verify = verify_none
fail_if_no_peer_cert = false
}
}

Thanks & Regards,
Mohammed.Raheem

**Comment by lafirest** [comment, marked as answer]
@raheem071989 
Hi, the official documentation is really helpful, please take a look [here](https://www.emqx.io/docs/en/latest/access-control/authn/authn.html).
You can see that for authentication there are many way to set up it, and the easiest method is using the [built-in database](https://www.emqx.io/docs/en/latest/access-control/authn/mnesia.html).
